### PR TITLE
Port resources file changes to 2.0.0

### PR DESCRIPTION
**Pull request by safern**
Per @shmao request I'm porting the resources extraction related work. 

Is this all that you need? Or did you only need partial port of this? 

cc: @weshaggard

### CODE REVIEW

**Comment by safern** [comment]
cc: @weshaggard

File: src/Microsoft.DotNet.Build.Tasks/project.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
     "NuGet.Versioning": "4.3.0-preview2-4095",
     "System.Diagnostics.TraceSource": "4.0.0",
     "System.Dynamic.Runtime": "4.0.11",
+    "System.IO.UnmanagedMemoryStream": "4.3.0",
```
Comment (weshaggard) [inline src/Microsoft.DotNet.Build.Tasks/project.json]: I wonder if this pulls up the versions of the other packages to the 4.3 set? I don't know if that will hurt anything but something to keep in mind if something goes wrong.

**Comment by weshaggard** [comment]
Looks reasonable. @safern were there any conflict resolutions you want me to look at in particular or was it straight forward?

**Comment by safern** [comment]
> Looks reasonable. @safern were there any conflict resolutions you want me to look at in particular or was it straight forward?

It was very straight forward, no conflicts at all.

**Pull request merged**